[UI-side compositing] Create RemoteScrollbarsController for plumbing web process events to the scrolling coordinator

nmoucht opened the pull request:
#### 87496bbd2fe0644663cbe1fb501e2711eddae3c3
<pre>
[UI-side compositing] Create RemoteScrollbarsController for plumbing web process events to the scrolling coordinator
<a href="https://bugs.webkit.org/show_bug.cgi?id=255173">https://bugs.webkit.org/show_bug.cgi?id=255173</a>
rdar://107770765

Reviewed by Simon Fraser.

This patch creates RemoteScrollbarsController, for the purpose of moving the scrolling coordinator
calls out of EventHandler and into the ScrollbarController functions, which were already being
called in the EventHandler.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::notifyScrollableAreasOfMouseEvents):
(WebCore::EventHandler::updateLastScrollbarUnderMouse):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::setMouseIsOverContentArea):
(WebCore::AsyncScrollingCoordinator::setMouseMovedInContentArea):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
(WebCore::ScrollingCoordinator::setMouseIsOverContentArea):
(WebCore::ScrollingCoordinator::setMouseMovedInContentArea):
* Source/WebCore/platform/ScrollbarsController.h:
(WebCore::ScrollbarsController::contentAreaWillPaint const):
(WebCore::ScrollbarsController::mouseEnteredContentArea):
(WebCore::ScrollbarsController::mouseExitedContentArea):
(WebCore::ScrollbarsController::mouseMovedInContentArea):
(WebCore::ScrollbarsController::mouseEnteredScrollbar const):
(WebCore::ScrollbarsController::mouseExitedScrollbar const):
(WebCore::ScrollbarsController::mouseIsDownInScrollbar const):
(WebCore::ScrollbarsController::willStartLiveResize):
(WebCore::ScrollbarsController::contentsSizeChanged const):
(WebCore::ScrollbarsController::willEndLiveResize):
(WebCore::ScrollbarsController::contentAreaDidShow):
(WebCore::ScrollbarsController::contentAreaDidHide):
(WebCore::ScrollbarsController::lockOverlayScrollbarStateToHidden):
(WebCore::ScrollbarsController::scrollbarsCanBeActive const):
(WebCore::ScrollbarsController::didAddVerticalScrollbar):
(WebCore::ScrollbarsController::willRemoveVerticalScrollbar):
(WebCore::ScrollbarsController::didAddHorizontalScrollbar):
(WebCore::ScrollbarsController::willRemoveHorizontalScrollbar):
(WebCore::ScrollbarsController::invalidateScrollbarPartLayers):
(WebCore::ScrollbarsController::verticalScrollbarLayerDidChange):
(WebCore::ScrollbarsController::horizontalScrollbarLayerDidChange):
(WebCore::ScrollbarsController::shouldScrollbarParticipateInHitTesting):
(WebCore::ScrollbarsController::horizontalScrollbarStateForTesting const):
(WebCore::ScrollbarsController::verticalScrollbarStateForTesting const):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createScrollingCoordinator const):
(WebKit::WebChromeClient::createScrollbarsController const):
* Source/WebKit/WebProcess/WebPage/mac/ScrollbarsControllerProxyMac.cpp: Added.
(WebKit::ScrollbarsControllerProxyMac::ScrollbarsControllerProxyMac):
(WebKit::ScrollbarsControllerProxyMac::mouseEnteredContentArea):
(WebKit::ScrollbarsControllerProxyMac::mouseExitedContentArea):
(WebKit::ScrollbarsControllerProxyMac::mouseMovedInContentArea):
(WebKit::ScrollbarsControllerProxyMac::mouseEnteredScrollbar const):
(WebKit::ScrollbarsControllerProxyMac::mouseExitedScrollbar const):
* Source/WebKit/WebProcess/WebPage/mac/ScrollbarsControllerProxyMac.h: Added.

Canonical link: <a href="https://commits.webkit.org/262744@main">https://commits.webkit.org/262744@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/737aaf84e108ade09b2ee0917ce835eebe03999e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2453 "26 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2452 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2575 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/3815 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2489 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2418 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2592 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2540 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2169 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2474 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/2255 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2206 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/3592 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/223 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2193 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/2049 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2323 "Built successfully") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2209 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3372 "run-api-tests-without-change (failure)") | 
| | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2243 "Passed tests") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2233 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2196 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2205 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/284 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2358 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/WebPage/mac/ScrollbarsControllerProxyMac.h:
```diff
@@ -0,0 +1,56 @@
+namespace WebKit {
+
+
+class ScrollbarsControllerProxyMac final : public WebCore::ScrollbarsController {
```
I don't think `ScrollbarsControllerProxyMac` is the right name for this; it's not a UI-process proxy of a `ScrollbarsController`, it's just an implementation specific to UI-side compositing. So it should follow the naming convention of `RemoteScrollingCoordinator`, and be `WebPage/RemoteLayerTree/RemoteScrollbarsController.h`.

nmoucht replied:
Right, fixed.

smfr reviewed (approved):
r=me with the move/rename.

nmoucht committed:
[UI-side compositing] Create RemoteScrollbarsController for plumbing web process events to the scrolling coordinator
https://bugs.webkit.org/show_bug.cgi?id=255173
rdar://107770765

Reviewed by Simon Fraser.

This patch creates RemoteScrollbarsController, for the purpose of moving the scrolling coordinator
calls out of EventHandler and into the ScrollbarController functions, which were already being
called in the EventHandler.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::notifyScrollableAreasOfMouseEvents):
(WebCore::EventHandler::updateLastScrollbarUnderMouse):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::setMouseIsOverContentArea):
(WebCore::AsyncScrollingCoordinator::setMouseMovedInContentArea):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
(WebCore::ScrollingCoordinator::setMouseIsOverContentArea):
(WebCore::ScrollingCoordinator::setMouseMovedInContentArea):
* Source/WebCore/platform/ScrollbarsController.h:
(WebCore::ScrollbarsController::contentAreaWillPaint const):
(WebCore::ScrollbarsController::mouseEnteredContentArea):
(WebCore::ScrollbarsController::mouseExitedContentArea):
(WebCore::ScrollbarsController::mouseMovedInContentArea):
(WebCore::ScrollbarsController::mouseEnteredScrollbar const):
(WebCore::ScrollbarsController::mouseExitedScrollbar const):
(WebCore::ScrollbarsController::mouseIsDownInScrollbar const):
(WebCore::ScrollbarsController::willStartLiveResize):
(WebCore::ScrollbarsController::contentsSizeChanged const):
(WebCore::ScrollbarsController::willEndLiveResize):
(WebCore::ScrollbarsController::contentAreaDidShow):
(WebCore::ScrollbarsController::contentAreaDidHide):
(WebCore::ScrollbarsController::lockOverlayScrollbarStateToHidden):
(WebCore::ScrollbarsController::scrollbarsCanBeActive const):
(WebCore::ScrollbarsController::didAddVerticalScrollbar):
(WebCore::ScrollbarsController::willRemoveVerticalScrollbar):
(WebCore::ScrollbarsController::didAddHorizontalScrollbar):
(WebCore::ScrollbarsController::willRemoveHorizontalScrollbar):
(WebCore::ScrollbarsController::invalidateScrollbarPartLayers):
(WebCore::ScrollbarsController::verticalScrollbarLayerDidChange):
(WebCore::ScrollbarsController::horizontalScrollbarLayerDidChange):
(WebCore::ScrollbarsController::shouldScrollbarParticipateInHitTesting):
(WebCore::ScrollbarsController::horizontalScrollbarStateForTesting const):
(WebCore::ScrollbarsController::verticalScrollbarStateForTesting const):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createScrollingCoordinator const):
(WebKit::WebChromeClient::createScrollbarsController const):
* Source/WebKit/WebProcess/WebPage/mac/ScrollbarsControllerProxyMac.cpp: Added.
(WebKit::ScrollbarsControllerProxyMac::ScrollbarsControllerProxyMac):
(WebKit::ScrollbarsControllerProxyMac::mouseEnteredContentArea):
(WebKit::ScrollbarsControllerProxyMac::mouseExitedContentArea):
(WebKit::ScrollbarsControllerProxyMac::mouseMovedInContentArea):
(WebKit::ScrollbarsControllerProxyMac::mouseEnteredScrollbar const):
(WebKit::ScrollbarsControllerProxyMac::mouseExitedScrollbar const):
* Source/WebKit/WebProcess/WebPage/mac/ScrollbarsControllerProxyMac.h: Added.

Canonical link: https://commits.webkit.org/262744@main
